File: sdk/clojure/src/hitechcloud/sdk/partner.clj
(ns hitechcloud.sdk.partner
  (:require [hitechcloud.sdk.http-client :as http]))

;; Partner: partner program, tiers, pricing, customers, earnings, wallet, payouts, leads, referral, rates

(defn get_partner_info "GET /api/partner - Get partner info" [client & [params]] (http-get client "/api/partner" params))

(defn apply "POST /api/partner/apply - Apply to partner program" [client & [params]] (http-post client "/api/partner/apply" params))

(defn list_tiers "GET /api/partner/tiers - List partner tiers" [client & [params]] (http-get client "/api/partner/tiers" params))

(defn get_pricing "GET /api/partner/pricing - Get partner pricing" [client & [params]] (http-get client "/api/partner/pricing" params))

(defn list_customers "GET /api/partner/customers - List customers" [client & [params]] (http-get client "/api/partner/customers" params))

(defn get_earnings "GET /api/partner/earnings - Get earnings" [client & [params]] (http-get client "/api/partner/earnings" params))

(defn get_wallet "GET /api/partner/wallet - Get wallet balance" [client & [params]] (http-get client "/api/partner/wallet" params))

(defn list_payouts "GET /api/partner/payouts - List payouts" [client & [params]] (http-get client "/api/partner/payouts" params))

(defn request_payout "POST /api/partner/payouts - Request payout" [client & [params]] (http-post client "/api/partner/payouts" params))

(defn list_leads "GET /api/partner/leads - List leads" [client & [params]] (http-get client "/api/partner/leads" params))

(defn create_lead "POST /api/partner/leads - Create lead" [client & [params]] (http-post client "/api/partner/leads" params))

(defn get_referral "GET /api/partner/referral - Get referral info" [client & [params]] (http-get client "/api/partner/referral" params))

(defn get_rates "GET /api/partner/rates - Get exchange rates" [client & [params]] (http-get client "/api/partner/rates" params))

File: sdk/fsharp/src/HiTechCloudSDK/Partner.fs
namespace HiTechCloud.SDK

/// Partner: partner program, tiers, pricing, customers, earnings, wallet, payouts, leads, referral, rates
type PartnerResource(httpClient: HttpClient) =

    /// GET /api/partner - Get partner info
    member this.GetPartnerInfo() =
        async {
            return! httpClient.getAsync("/api/partner")
        }

    /// POST /api/partner/apply - Apply to partner program
    member this.Apply() =
        async {
            return! httpClient.postAsync("/api/partner/apply")
        }

    /// GET /api/partner/tiers - List partner tiers
    member this.ListTiers() =
        async {
            return! httpClient.getAsync("/api/partner/tiers")
        }

    /// GET /api/partner/pricing - Get partner pricing
    member this.GetPricing() =
        async {
            return! httpClient.getAsync("/api/partner/pricing")
        }

    /// GET /api/partner/customers - List customers
    member this.ListCustomers() =
        async {
            return! httpClient.getAsync("/api/partner/customers")
        }

    /// GET /api/partner/earnings - Get earnings
    member this.GetEarnings() =
        async {
            return! httpClient.getAsync("/api/partner/earnings")
        }

    /// GET /api/partner/wallet - Get wallet balance
    member this.GetWallet() =
        async {
            return! httpClient.getAsync("/api/partner/wallet")
        }

    /// GET /api/partner/payouts - List payouts
    member this.ListPayouts() =
        async {
            return! httpClient.getAsync("/api/partner/payouts")
        }

    /// POST /api/partner/payouts - Request payout
    member this.RequestPayout() =
        async {
            return! httpClient.postAsync("/api/partner/payouts")
        }

    /// GET /api/partner/leads - List leads
    member this.ListLeads() =
        async {
            return! httpClient.getAsync("/api/partner/leads")
        }

    /// POST /api/partner/leads - Create lead
    member this.CreateLead() =
        async {
            return! httpClient.postAsync("/api/partner/leads")
        }

    /// GET /api/partner/referral - Get referral info
    member this.GetReferral() =
        async {
            return! httpClient.getAsync("/api/partner/referral")
        }

    /// GET /api/partner/rates - Get exchange rates
    member this.GetRates() =
        async {
            return! httpClient.getAsync("/api/partner/rates")
        }

